server/pulp/server/controllers/repository.py

```diff
@@ -789,16 +780,22 @@ def sync(repo_id, sync_config_override=None, scheduled_call_id=None):
         else:
             msg = _('Plugin type [%s] on repo [%s] did not return a valid sync report')
             _logger.warn(msg % (repo_importer['importer_type_id'], repo_obj.repo_id))
-            added_count = updated_count = removed_count = -1  # None?
             summary = details = msg
             result_code = RepoSyncResult.RESULT_ERROR  # RESULT_UNKNOWN?
 
-        sync_result = RepoSyncResult.expected_result(
-            repo_obj.repo_id, repo_importer['id'], repo_importer['importer_type_id'],
-            sync_start_timestamp, sync_end_timestamp, added_count, updated_count, removed_count,
-            summary, details, result_code)
+        sync_result, sync_end_timestamp = _reposync_result(repo_obj, repo_importer,
+                                                           sync_start_timestamp, summary, details,
+                                                           result_code, before_sync_unit_count)
 
     finally:
+        if sync_result is None:
```

Comment:
Why do we have to handle this case? I expect that if cancel occurs the task and all of its threads will exit immediately.